yt_dlp/extractor/msn.py

```diff
@@ -1,167 +1,208 @@
+            'modified_timestamp': 1740488561,
+            'modified_date': '20250225',
+        },
+        'playlist_count': 1,
     }]
 
     def _real_extract(self, url):
-        display_id, page_id = self._match_valid_url(url).groups()
-
-        webpage = self._download_webpage(url, display_id)
-
-        entries = []
-        for _, metadata in re.findall(r'data-metadata\s*=\s*(["\'])(?P<data>.+?)\1', webpage):
-            video = self._parse_json(unescapeHTML(metadata), display_id)
+        locale, display_id, page_id = self._match_valid_url(url).group('locale', 'display_id', 'id')
 
-            provider_id = video.get('providerId')
-            player_name = video.get('playerName')
-            if player_name and provider_id:
-                entry = None
-                if player_name == 'AOL':
-                    if provider_id.startswith('http'):
-                        provider_id = self._search_regex(
-                            r'https?://delivery\.vidible\.tv/video/redirect/([0-9a-f]{24})',
-                            provider_id, 'vidible id')
-                    entry = self.url_result(
-                        'aol-video:' + provider_id, 'Aol', provider_id)
-                elif player_name == 'Dailymotion':
-                    entry = self.url_result(
-                        'https://www.dailymotion.com/video/' + provider_id,
-                        'Dailymotion', provider_id)
-                elif player_name == 'YouTube':
-                    entry = self.url_result(
-                        provider_id, 'Youtube', provider_id)
-                elif player_name == 'NBCSports':
-                    entry = self.url_result(
-                        'http://vplayer.nbcsports.com/p/BxmELC/nbcsports_embed/select/media/' + provider_id,
-                        'NBCSportsVPlayer', provider_id)
-                if entry:
-                    entries.append(entry)
-                    continue
+        json_data = self._download_json(
+            f'https://assets.msn.com/content/view/v2/Detail/{locale}/{page_id}', page_id,
+            note='Downloading JSON metadata')
 
-            video_id = video['uuid']
-            title = video['title']
+        common_metadata = traverse_obj(json_data, {
+            'title': ('title', {str}),
+            'description': (('abstract', ('body', {clean_html})), {str}, filter, any),
+            'timestamp': ('createdDateTime', {parse_iso8601}),
+            'release_timestamp': ('publishedDateTime', {parse_iso8601}),
+            'modified_timestamp': ('updatedDateTime', {parse_iso8601}),
+            'thumbnail': ('thumbnail', 'image', 'url', {url_or_none}),
+            'duration': ('videoMetadata', 'playTime', {int_or_none}),
+            'tags': ('keywords', ..., {str}),
+            'uploader': ('provider', 'name', {str}),
+            'uploader_id': ('provider', 'id', {str}),
+        })
 
+        page_type = traverse_obj(json_data, ('type', {str}))
```

Comment:
I think we can have this be fatal since we raise anyways if it's not a known string value
```suggestion
        page_type = json_data['type']
```